File: Tome/tome-back/src/main/kotlin/tome/books/Book.kt
package tome.books

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany
import jakarta.persistence.ManyToOne
import tome.books.author.Author
import tome.books.languageApi.Language
import tome.books.publisherApi.Publisher
import tome.books.tag.Tag
import java.time.LocalDate
import java.util.UUID

@Entity
class Book(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = UUID.randomUUID(),
    val title: String,
    val isbn: String,
    val releasedDate: LocalDate,
    val pages: Int,
    val coverUrl: String,
    @Column(columnDefinition = "TEXT", length = 5000)
    val synopsis: String,
    @ManyToOne
    val language: Language,
    @ManyToOne
    val publisher: Publisher,
    @ManyToMany
    val tags: List<Tag>,
    @ManyToMany
    val authors: List<Author>,
)

File: Tome/tome-ui/lib/types/Book.ts
class Book {
  public id?: string;
  public title: string;
  public author: string;
  public isbn?: string;
  public pages?: number;
  public publishedDate?: Date;
  public coverUrl?: string;
  public description?: string;
  public genre?: string[];
  public rating?: number;

  constructor(
    title: string,
    author: string,
    isbn?: string,
    publishedDate?: Date,
    pages?: number,
    coverUrl?: string,
    description?: string,
    genre?: string[],
  ) {
    this.title = title;
    this.author = author;
    this.isbn = isbn;
    this.publishedDate = publishedDate;
    this.pages = pages;
    this.coverUrl = coverUrl;
    this.description = description;
    this.genre = genre || [];
    this.rating = 0;
  }
}
